pkg/activator/handler/handler.go

```diff
@@ -128,8 +129,17 @@ func (a *activationHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		}, "ThrottlerTry")
 		trySpan.End()
 
-		if err == activatornet.ErrActivatorOverload {
-			http.Error(w, activatornet.ErrActivatorOverload.Error(), http.StatusServiceUnavailable)
+		var errmsg string
+		switch err {
+		case activatornet.ErrActivatorOverload:
+			errmsg = err.Error()
+			fallthrough
+		case queue.ErrContextDone:
+			errmsg = "Timed out"
```

Comment:
This shouldn't be the business of this logic. The context can be either canceled or timed out. The err will tell (see comment below)

Reply:
We filter the error type so we only return known safe errors to users, so there does need to be some logic here around type. Updated to be less incorrect though.